### PR TITLE
Major Overhaul: Scope of Core & Simplification

### PR DESCRIPTION
**Pull request by henkbirkholz**

### CODE REVIEW

File: draft-ietf-rats-corim.md [hunk 1]
```diff
@@ -121,37 +119,93 @@ informative:
...
 Not only Attesters can evolve and therefore new measurement types need to be expressed, but an Endorser may also want to provide new security relevant attributes about an Attester at a future point in time.
 
-This document specifies Concise Reference Integrity Manifests (CoRIM) a CBOR {{-cbor}} based data model addressing the above challenges by using an extensible format common to all supply chain actors and Verifiers.
+This document specifies Concise Reference Integrity Manifests (CoRIM); a CBOR {{-cbor}} based data model addressing the above challenges by using an extensible format common to all supply chain actors and Verifiers.
 CoRIM enables Verifiers to reconcile a complex and scattered supply chain into a single homogeneous view.
 
+## Verifier Reconciliation {#sec-verifier-rec}
+
+While this specification focusses on the CoRIM format itself, understanding the procedures how Verifiers are assumed to digest CoRIM documents provides vital context.
+A Verifier hast to reconcile its various inputs, CoRIM being one of them.
```
Comment (henkbirkholz) [inline draft-ietf-rats-corim.md:153]:
```suggestion
A Verifier has to reconcile its various inputs, CoRIM being one of them.
```

**Comment by nedmsmith** [comment]
I added sections at the beginning of the appraisal description section. However, the existing text that anticipates using a different internal representation still needs attention.

**Review by nedmsmith — APPROVED** [review]
LGTM

**Review by nedmsmith — APPROVED** [review]
LGTM

File: draft-ietf-rats-corim.md [hunk 2]
```diff
@@ -1514,6 +1514,58 @@ CoRIM-based data structures define an external representation of Conceptual Mess
...
+### **Phase 3**: Reference Values Corroboration and Augmentation
+
+During Phase 3, Reference Values inputs are compared with Evidence inputs.
+Reference Values inputs describe possible states of Attesters.
+If the actual state of the Attester is described by the possible Attester states, then the overlapping (corroborated) actual states are added to the Attester's actual state.
+These inputs are added with the Reference Value Provider's authority.
+
+### **Phase 4**: Endorsed Values Augmentation
+
+During Phase 4, Endorsed Values inputs containing conditions that describe expected Attester state are compared.
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1548]: Compared against whom ?

File: draft-ietf-rats-corim.md [hunk 3]
```diff
@@ -1546,25 +1546,26 @@ These inputs are added with the Reference Value Provider's authority.
 ### **Phase 4**: Endorsed Values Augmentation
 
 During Phase 4, Endorsed Values inputs containing conditions that describe expected Attester state are compared.
-If the comparison is satisfied, then additional Claims about the Attester are added.
+If the comparison is satisfied, then additional Claims about the Attester are added to the Accepted Claim Set.
```
Comment (henkbirkholz) [inline draft-ietf-rats-corim.md:1549]:
```suggestion
If the comparison is satisfied, then additional Claims about the Attester are added to the Actual-State Claim Set.
```

File: draft-ietf-rats-corim.md (hunk 2 above)
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md]: Addressed in [4857331](https://github.com/ietf-rats-wg/draft-ietf-rats-corim/pull/232/commits/48573314c87a73fda94200a755c207e2efd21ec5)

**Comment by henkbirkholz** [comment]
ECT are not actually tuples anymore (even if we agree that authority moves into environment), right?

File: draft-ietf-rats-corim.md [hunk 4]
```diff
@@ -163,148 +158,129 @@ and {{Section G of -cddl}}.
...
-
-~~~ cddl
-{::include cddl/ueid.cddl}
+ACS = {
+  &(actual-state: 0) => [ + ECT ]
+}
 ~~~
 
-### OID {#sec-common-oid}
+In this document, CDDL definitions are used to illustrate examples for internal representations.
```
Comment (henkbirkholz) [inline draft-ietf-rats-corim.md:233]: Isn't this going a bit far? This document definitely comes with CDDL for CoRIM in CBOR, right?

File: draft-ietf-rats-corim.md (hunk 3 above)
Comment (yogeshbdeshpande) [reply draft-ietf-rats-corim.md:1549]: Bulk of the document states ACS as Accepted Claims Set.  We need to agree on a common term and replace the entire document for consistency!

Comment (deeglaze) [reply draft-ietf-rats-corim.md:1549]: Indeed "actual" is more philosophical. We're only accepting an authority's attestation of actual state. Ceci n'est pas une pipe.

**Review by deeglaze — CHANGES_REQUESTED** [review]

File: draft-ietf-rats-corim.md (hunk 3 above)
Comment (nedmsmith) [reply draft-ietf-rats-corim.md:1550]: I'm OK with "Actual-state Claims Set" but it should be applied consistently. I prefer using ACS everywhere and define ACS in one place. Then it's easy to make a global change.

**Comment by nedmsmith** [comment]
> ECT are not actually tuples anymore (even if we agree that authority moves into environment), right?

It's a 4-tuple or 5-tuple instead of a 2-tuple. But a tuple nevertheless.

File: draft-ietf-rats-corim.md [hunk 5]
```diff
@@ -1409,27 +1383,191 @@ The following describes each member of the `concise-bom-tag` map.
...
+
+Inputs to a Verifier are mapped from their external representation to an internal representation.
+CoRIM defines structure for Conceptual Messages that includes Endorsements and Reference Values.
+CoRIM data structures may be used by Evidence and Attestation Results that wish to describe overlapping structure.
+CoRIM-based data structures define an external representation of Conceptual Messages that are mapped to an internal representation.
+Appraisal processing describes both mapping transformations and Verifier reconciliation {{sec-verifier-rec}}.
+Non-CoRIM-based data structures require mapping transformation, but these are out of scope for this document.
+
+If a CoRIM profile is specified, there are a few well-defined points in the procedure where Verifier behaviour
+depends on the profile.
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:1522]: fixed

File: draft-ietf-rats-corim.md [hunk 6]
```diff
@@ -1496,67 +1770,79 @@ In such case, the Verifier Owner may instruct the Verifier to discard tags activ
...
+Evidence is divided up into one or more `ev` relations where the `condition` ECT identifies the Attester from which Evidence was collected. If the Verifier maintains multiple Attester sessions, the Verifier session may be identified using an ECT.
+
+Evidence information is mapped to an `addition` ECT that populates each of the ECT fields. If the Evidence doesn't have a value for the mandatory fields, the Verifier MUST NOT process the Evidence.
+
+The `e` field is populated with a Target Environment identifier.
+The `c` field is populated with the measurements collected by an Attesting Environment.
+The `a` field is populated with the identity of the entity that asserted (e.g., signed) the Evidence.
+The `ns` field is populated with the namespace context if supplied. For example, the Attester's manufacturer may have a URI that identifies the manufacturing series, family or architecture.
+
+Evidence transformation algorithms may be well-known, see {{sec-spdm}} and {{sec-dice}};
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md]: no, not a cause for a line break
in simple kramdown `;` is just a character used in sentences

File: draft-ietf-rats-corim.md [hunk 7]
```diff
@@ -1496,67 +1770,79 @@ In such case, the Verifier Owner may instruct the Verifier to discard tags activ
...
+
+## Evidence Appraisal {#sec-ev-appraisal}
+
+### The Acepted Claims Set
 
 At the end of the Evidence collection process Evidence has been converted into
-a format suitable for appraisal. To this end, this document describes an
-`accepted-claims-set` format and the algorithms used to compare it against
-CoMID Reference Values.
+an internal represenetation suitable for appraisal. See {{sec-ir-cm}}.
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md]: fixed

File: draft-ietf-rats-corim.md [hunk 8]
```diff
@@ -1496,67 +1770,79 @@ In such case, the Verifier Owner may instruct the Verifier to discard tags activ
...
+Evidence information is mapped to an `addition` ECT that populates each of the ECT fields. If the Evidence doesn't have a value for the mandatory fields, the Verifier MUST NOT process the Evidence.
+
+The `e` field is populated with a Target Environment identifier.
+The `c` field is populated with the measurements collected by an Attesting Environment.
+The `a` field is populated with the identity of the entity that asserted (e.g., signed) the Evidence.
+The `ns` field is populated with the namespace context if supplied. For example, the Attester's manufacturer may have a URI that identifies the manufacturing series, family or architecture.
+
+Evidence transformation algorithms may be well-known, see {{sec-spdm}} and {{sec-dice}};
+may be defined by a CoRIM profile ({{sec-corim-profile-types}});
+or may be supplied dynamically. The handling of dynamic Evidence transformation algorithms is out of scope for this document.
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md]: fixed

File: draft-ietf-rats-corim.md [hunk 9]
```diff
@@ -1496,67 +1770,79 @@ In such case, the Verifier Owner may instruct the Verifier to discard tags activ
...
-a format suitable for appraisal. To this end, this document describes an
-`accepted-claims-set` format and the algorithms used to compare it against
-CoMID Reference Values.
+an internal represenetation suitable for appraisal. See {{sec-ir-cm}}.
 
-~~~ cddl
-{::include cddl/accepted-claims-set.cddl}
-~~~
+Verifiers are not required to use this as their internal representation,
+for the purposes of this document appraisal is described in terms of the above cited internal representation.
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md]: fixed

File: draft-ietf-rats-corim.md [hunk 10]
```diff
@@ -2146,6 +2387,52 @@ Environments (CoRE) Parameters" Registry {{!IANA.core-parameters}}:
 
 --- back
 
+# Transforming SPDM Evidence {#sec-spdm}
+
+This section defines how Evidence from SPDM {{SPDM}} is transformed into a
+format where it can be added to an accepted claims set.
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:2443]: fixed

File: draft-ietf-rats-corim.md [hunk 11]
```diff
@@ -2146,6 +2387,52 @@ Environments (CoRE) Parameters" Registry {{!IANA.core-parameters}}:
...
+A Verifier supporting SPDM format Evidence should implement this section.
+
+The TCG DICE Concise Evidence Binding for SPDM specification {{CE.SPDM}} describes the process by which
+measurements in an SPDM Measurement Block are converted to Evidence suitable for
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:2447]: fixed

File: draft-ietf-rats-corim.md [hunk 12]
```diff
@@ -2146,6 +2387,52 @@ Environments (CoRE) Parameters" Registry {{!IANA.core-parameters}}:
...
+matching using the rules below.
+The SPDM measurements are converted to `concise-evidence` which has a format that is similar
+to CoRIM `triples-map` (their semantics follows the matching rules described above).
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:2450]: fixed

File: draft-ietf-rats-corim.md [hunk 13]
```diff
@@ -2146,6 +2387,52 @@ Environments (CoRE) Parameters" Registry {{!IANA.core-parameters}}:
...
+
+# Transforming DICE Evidence {#sec-dice}
+
+This section defines how Evidence from DICE {{DICE.AA}} is transformed into a
+format where it can be added to an accepted claims set.
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:2455]: fixed

File: draft-ietf-rats-corim.md [hunk 14]
```diff
@@ -2146,6 +2387,52 @@ Environments (CoRE) Parameters" Registry {{!IANA.core-parameters}}:
...
+A Verifier supporting DICE format Evidence should implement this section.
+
+DICE Evidence appears in certificates in the TcbInfo or MultiTcbInfo extension.
+Each TcbInfo, and each entry in the MultiTcbInfo, is converted to an
+`endorsed-triple-record` using the rules in this section.
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:2460]: fixed

File: draft-ietf-rats-corim.md [hunk 15]
```diff
@@ -2146,6 +2387,52 @@ Environments (CoRE) Parameters" Registry {{!IANA.core-parameters}}:
...
+In a MultiTcbInfo each entry in the sequence is treated as independent and
+translated into a separate Evidence object.
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:2462]: fixed

[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:2462]: fixed

File: draft-ietf-rats-corim.md [hunk 16]
```diff
@@ -2146,6 +2387,52 @@ Environments (CoRE) Parameters" Registry {{!IANA.core-parameters}}:
...
+
+The Verifier SHALL translate each field in the TcbInfo into a field in the
+created endorsed-triple-record
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:2465]: fixed

File: draft-ietf-rats-corim.md [hunk 17]
```diff
@@ -2146,6 +2387,52 @@ Environments (CoRE) Parameters" Registry {{!IANA.core-parameters}}:
...
+- The TcbInfo `fwids` field SHALL be translated to the field named `measurement-map / mval / digests`
+  - Each digest within fwids is translated to a CoMID digest object, with an appropriate algorithm identifier
+- The TcbInfo `flags` field SHALL be translated to the field named `measurement-map / mval / flags`
+  - Each flag is translated independently
+- The TcbInfo `vendorInfo` SHALL shall be copied to the field named `measurement-map / mval / raw-value`
+
+If there are multiple `endorsed-triple-record`s with the same `environment-map`
+then they MUST be merged into a single entry.
+If the `measurement-values-map` fields in Evidence triples have conflicting
+values then the Verifier MUST fail validation.
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:2484]: fixed

File: draft-ietf-rats-corim.md [hunk 18]
```diff
@@ -2146,6 +2387,52 @@ Environments (CoRE) Parameters" Registry {{!IANA.core-parameters}}:
...
+- The TcbInfo `version` field SHALL be translated to the field named `measurement-map / mval / version / version`
+- The TcbInfo `svn` field SHALL be copied to the field named `measurement-map / mval / svn`
+- The TcbInfo `fwids` field SHALL be translated to the field named `measurement-map / mval / digests`
+  - Each digest within fwids is translated to a CoMID digest object, with an appropriate algorithm identifier
+- The TcbInfo `flags` field SHALL be translated to the field named `measurement-map / mval / flags`
+  - Each flag is translated independently
+- The TcbInfo `vendorInfo` SHALL shall be copied to the field named `measurement-map / mval / raw-value`
+
+If there are multiple `endorsed-triple-record`s with the same `environment-map`
+then they MUST be merged into a single entry.
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:2482]: fixed

File: draft-ietf-rats-corim.md [hunk 19]
```diff
@@ -1409,27 +1383,191 @@ The following describes each member of the `concise-bom-tag` map.
...
+An opaque, variable-length byte string.
+It can be used in different contexts: as an instance, class or group identifier in an `environment-map`; as a raw value measurement in a `measurement-values-map`.
+Its semantics are defined by the context in which it is found, and by the overarching CoRIM profile.
+When used as an identifier the responsible allocator entity SHOULD ensure uniqueness within the context that it is used.
+
+~~~ cddl
+{::include cddl/tagged-bytes.cddl}
+~~~
+
+# Appraisal of CoRIM-based Inputs {#sec-appr-corim-inputs}
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:1512]: CMW is an umbrella structure ofr all types of RATS Conceptual Messages. You could - I am making things up here - bundle TPM Evidence, EAT Attestation Results, and a CoRIM in a CMW (each with their own media type).

File: draft-ietf-rats-corim.md [hunk 20]
```diff
@@ -1409,27 +1383,191 @@ The following describes each member of the `concise-bom-tag` map.
...
+
+Inputs to a Verifier are mapped from their external representation to an internal representation.
+CoRIM defines structure for Conceptual Messages that includes Endorsements and Reference Values.
+CoRIM data structures may be used by Evidence and Attestation Results that wish to describe overlapping structure.
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:1516]: I do not want to loose this concern/comment when we merge the PR. @deeglaze Do you want to rescue this comment into a better visible issue?

**Comment by henkbirkholz** [comment]
> ECT are not actually tuples anymore (even if we agree that authority moves into environment), right?

Over-obsessing on "tuple". Tuple and therefore ECT is fine (independent of number of items tuple'ed).

File: draft-ietf-rats-corim.md [hunk 21]
```diff
@@ -1837,63 +2194,28 @@ If no entry is found, the Reference Value does not match.
...
-- The TcbInfo `index` field SHALL be copied to the field named `environment-map / class / index`
-
-- The TcbInfo `version` field SHALL be translated to the field named `measurement-map / mval / version / version`
-- The TcbInfo `svn` field SHALL be copied to the field named `measurement-map / mval / svn`
-- The TcbInfo `fwids` field SHALL be translated to the field named `measurement-map / mval / digests`
-  - Each digest within fwids is translated to a CoMID digest object, with an appropriate algorithm identifier
-- The TcbInfo `flags` field SHALL be translated to the field named `measurement-map / mval / flags`
-  - Each flag is translated independently
-- The TcbInfo `vendorInfo` SHALL shall be copied to the field named `measurement-map / mval / raw-value`
+A profile may specify handling for new CBOR tagged Reference Values. The
```
Comment (deeglaze) [inline draft-ietf-rats-corim.md:2213]:
```suggestion
A profile may specify handling for new CBOR tagged Reference Values.
The profile must specify how to compare the CBOR tagged Reference Value against the ACS.
```

**Review by yogeshbdeshpande — APPROVED** [review]
LGMT!

**Review by deeglaze — APPROVED** [review]

File: draft-ietf-rats-corim.md [hunk 22]
```diff
@@ -111,59 +109,117 @@ informative:
...
+Complementary to the external CoRIM documents, a Verifier is assumed to construct an internal representation of each input and map each external representation to an internal representation.
+Using the internal representation, the Verifier processes inputs as though they are parts of a conversation where the Verifier keeps track of who said what.
+Namely, the provenance of the inputs is tracked as *authority*.
+In the case of CoRIM, the CoRIM issuer identifies its authority.
+
+Effectively, Attesters, Reference Value Providers, Endorsers, Verifier Owners, Relying Parties, and even the Verifier potentially all contribute to the conversation.
+Each producer of corresponding RATS Conceptual Messages can assert Claims about an Attester's actual or allowed state.
+The Verifier's objective is to produce a list of Claims that describe the Attester's presumed actual state.
+Producers of RATS Conceptual Messages can assert contradictory assertions.
+For example, a compromised Attester may produce false assertions that create a conflict with an RVP's issued Reference Values.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:176]:
```suggestion
For example, a compromised Attester may produce false statements that conflict with the Reference Values provided by a Reference Value Provider (RVP).
```

File: draft-ietf-rats-corim.md [hunk 23]
```diff
@@ -111,59 +109,117 @@ informative:
...
+In essence, if Evidence is not corroborated by a Reference Value Provider's (RVP) assertions, then the RVP's Claims are not included in the internal representation of an Attester's actual state, a.k.a., Accepted Claims Set (ACS).
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:177]:
```suggestion
In essence, if Evidence is not corroborated by an RVP's assertions, then the RVP's Claims are not included in the internal representation of an Attester's actual state, a.k.a., Accepted Claims Set (ACS).
```

File: draft-ietf-rats-corim.md [hunk 24]
```diff
@@ -111,59 +109,117 @@ informative:
...
+
+> *[Ned] Suggest we use Actual-state Claims Set (ACS) since "accepted" implies an acceptance policy input.*
+
+A Verifier relies on input from appraisal policy to identify relevant assertions included in the ACS.
+For example, if a policy requires corroborated assertions issued by a particular RVP, then those assertions may be conveyed as Attestation Results.
+The Verifier may produce new assertions as a result of an applied appraisal policy.
+For example, if an appraisal procedure finds all of the components of a subsystem are configured correctly, the policy may direct the Verifier to produce new assertions, "Subsystem=X" has the Claim "TRUSTED=TRUE".
+Consequently, the internal ACS structure is a reconciled conversation between several producers of RATS Conceptual Messages that has mapped each message into a consistent internal representation, has associated the identity of the corresponding RATS role with each assertion (the authority), and has applied Conceptual Message constraints to the assertion.
+
+The CoRIM data model specified in this document covers the RATS Conceptual Message types, `Reference Values` and `Endorsements`.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:187]:
```suggestion
The CoRIM data model specified in this document covers the RATS Conceptual Message types, "Reference Values" and "Endorsements".
```

File: draft-ietf-rats-corim.md [hunk 25]
```diff
@@ -111,59 +109,117 @@ informative:
...
+Reference values and Endorsements are required for Verifier reconciliation (and Evidence is required for corresponding internal ACS creation) as illustrated in {{sec-interact-acs}}.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:188]:
```suggestion
Reference values and Endorsements are required for Verifier reconciliation, and Evidence is required for corresponding internal ACS creation as illustrated in {{sec-interact-acs}}.
```

File: draft-ietf-rats-corim.md [hunk 26]
```diff
@@ -111,59 +109,117 @@ informative:
...
+
+In this document CDDL is used to specify both the CoRIM structure and to specify an internal representation for use in the appraisal procedure.
+The actual internal representation of a Verifier is implementation-specific and out-of-scope of this document.
+Requirements for an internal representation of Conceptual Messages are defined in {{tbl-cmrr}}, where each Conceptual Message type has a structure as depicted by the *Structure* column.
+The internal representations used by this document are defined in {{sec-ir-cm}}.
+
+## Interacting with an ACS {#sec-interact-acs}
+
+Conceptual Messages interact with an ACS by specifying criteria that should be met by the ACS and by presenting the assertions that should be added to the ACS if the criteria are satisfied.
+Internal representations of Conceptual Messages, ACS, and ARS should satisfy the following requirements for Verifier reconciliation and appraisal processing:
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:200]: need to expand "ARS" on first use :-)

File: draft-ietf-rats-corim.md [hunk 27]
```diff
@@ -111,59 +109,117 @@ informative:
...
+
+| CM Type | Structure | Description |
+|---
+| Evidence | List of Evidence claims | If the Attester is authenticated, add Evidence claims to the ACS with Attester authority |
+| Reference Values | List of Reference Values claims | If a reference value in a CoRIM matches claims in the ACS, then the authority of the CoRIM issuer is added to those claims. |
+| Endorsements | List of expected actual state claims, List of Endorsed Values claims | If the list of expected claims are in the ACS, then add the list of Endorsed Values claims to the ACS with Endorser authority |
+| Series Endorsements | List of expected actual state claims, Series of tuples containing selection and series Endorsement claims | If the list of expected claims are in the ACS, and if the series selection condition is satisfied, then add the series Endorsement claims for the first series tuple with satisfied condition to the ACS with Endorser authority |
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:207]: unclear what "selection" means in this context

File: draft-ietf-rats-corim.md [hunk 28]
```diff
@@ -111,59 +109,117 @@ informative:
...
+| Verifier | List of expected actual state claims, List of Verifier-generated claims | If the list of expected claims are in the ACS, then add the list of Verifier-generated claims to the ACS with Verifier authority |
+| Policy | List of expected actual state claims, List of Policy-generated claims | If the list of expected claims are in the ACS, then add the list of Policy-generated claims to the ACS with Policy Owner authority |
+| Attestation Results | List of expected actual state claims, List of expected Attestation Results claims | If the list of expected claims are in the ACS, then copy the list of Attestation Results claims into the ARS |
+{: #tbl-cmrr title="Conceptual Message Representation Requirements"}
+
+## Quantizing Inputs {#sec-quantize}
+
+To be completed!
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:215]: add a reference to the tracking issue instead

File: draft-ietf-rats-corim.md [hunk 29]
```diff
@@ -111,200 +109,136 @@ informative:
...
 
-~~~ cddl
-{::include cddl/digest.cddl}
-~~~
+## Quantizing Inputs {#sec-quantize}
 
-A measurement can be obtained using different hash algorithms.
-A `digests-type` can be used to collect multiple digest values obtained by applying different hash algorithms on the same input.
-Each entry in the `digests-type` MUST have a unique `alg` value.
+Inputs may change soon after a Verifier processes them causing the Verifier to consider how often it should check for updated inputs.
```
[the remark this answers is not in the document]
Comment (thomas-fossati) [reply draft-ietf-rats-corim.md:216]: I agree with @deeglaze

**Review by thomas-fossati — COMMENTED** [review]
some late comments, apologies

File: draft-ietf-rats-corim.md (hunk 27 above)
Comment (nedmsmith) [reply draft-ietf-rats-corim.md:207]: It is a second stage condition. 
The series structure is a list of condition-addition 2-tuples. Selection is another word for condition that isn't confused with the endorsement's first-stage condition.
Possibly, the second sentence here is a clearer explanation of selection. The current text assumes the reader read ahead to understand it's use here.

Comment (thomas-fossati) [reply draft-ietf-rats-corim.md:207]: OK, thanks. The problem is this is not self-contained.  At least there should be a forward reference.

Comment (nedmsmith) [inline draft-ietf-rats-corim.md:207]:
```suggestion
| Series Endorsements | List of expected actual state claims and a series of selection-addition tuples | If the expected claims are in the ACS, and if the series selection condition is satisfied, then add the additional claims to the ACS with Endorser authority. See {{sec-ir-end-val}} |
```

**Review by nedmsmith — COMMENTED** [review]
See suggested change

File: draft-ietf-rats-corim.md (hunk 27 above)
Comment (thomas-fossati) [reply draft-ietf-rats-corim.md:207]: nice, thanks!

File: draft-ietf-rats-corim.md [hunk 30]
```diff
@@ -111,59 +109,117 @@ informative:
...
 in particular, CBOR diagnostic notation is defined in {{Section 8 of -cbor}}
-and {{Section G of -cddl}}.
+and {{Section G of -cddl}}. Terms and concepts are always referenced as proper nouns, i.e., with Capital Letters.
 
 {::boilerplate bcp14}
 
-## CDDL Typographical Conventions
+# Verifier Reconciliation {#sec-verifier-rec}
+
+While this specification focuses on the CoRIM format itself, it is vital to understand its intended meaning for the mechanics of how Verifiers should process CoRIM documents.
```
[the remark this answers is not in the document]
Comment (thomas-fossati) [reply draft-ietf-rats-corim.md:165]: Thanks for the attempt to clarify the sentence.

However, the document focuses on the description of the data format & the processing.  There is no primary vs secondary focus.

File: draft-ietf-rats-corim.md [hunk 31]
```diff
@@ -111,59 +109,117 @@ informative:
...
+A Verifier has to reconcile its various inputs, CoRIM being one of them.
+Complementary to the external CoRIM documents, a Verifier is assumed to construct an internal representation of each input and map each external representation to an internal representation.
+Using the internal representation, the Verifier processes inputs as though they are parts of a conversation where the Verifier keeps track of who said what.
+Namely, the provenance of the inputs is tracked as *authority*.
+In the case of CoRIM, the CoRIM issuer identifies its authority.
```
[the remark this answers is not in the document]
Comment (yogeshbdeshpande) [reply draft-ietf-rats-corim.md:170]: We agreed to keep the older version, of the text as there is no permanent external representation and New text 167 creates that `ambiguity`

[the remark this answers is not in the document]
Comment (thomas-fossati) [reply draft-ietf-rats-corim.md:170]: which "older version" are you referring to?

Both OLD and NEW have the same "external"...

File: draft-ietf-rats-corim.md (hunk 26 above)
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:200]:
```suggestion
Internal representations of Conceptual Messages, ACS, and Attestation Results Set (ARS) should satisfy the following requirements for Verifier reconciliation and appraisal processing:
```

Comment (yogeshbdeshpande) [reply draft-ietf-rats-corim.md:200]: Add a reference to ARS

File: draft-ietf-rats-corim.md [hunk 32]
```diff
@@ -111,59 +109,117 @@ informative:
...
+| Verifier | List of expected actual state claims, List of Verifier-generated claims | If the list of expected claims are in the ACS, then add the list of Verifier-generated claims to the ACS with Verifier authority |
+| Policy | List of expected actual state claims, List of Policy-generated claims | If the list of expected claims are in the ACS, then add the list of Policy-generated claims to the ACS with Policy Owner authority |
+| Attestation Results | List of expected actual state claims, List of expected Attestation Results claims | If the list of expected claims are in the ACS, then copy the list of Attestation Results claims into the ARS |
+{: #tbl-cmrr title="Conceptual Message Representation Requirements"}
+
+## Quantizing Inputs {#sec-quantize}
+
+To be completed!
+Inputs may change soon after a Verifier processes them causing the Verifier to consider how often it should check for updated inputs.
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:216]:
```suggestion

```

File: draft-ietf-rats-corim.md (hunk 28 above)
Comment (yogeshbdeshpande) [reply draft-ietf-rats-corim.md:215]: issue #242

File: draft-ietf-rats-corim.md (hunk 30 above)
[the remark this answers is not in the document]
Comment (thomas-fossati) [reply draft-ietf-rats-corim.md]: The corrected text fixes my previous concern.

**Comment by thomas-fossati** [comment]
The CI is broken.

Locally, this PR compiles fine.

We decided to merge it into main nevertheless to prioritise functionality over safety.

See #243

**Review by thomas-fossati — APPROVED** [review]

**Pull request merged**